src/vs/workbench/api/browser/viewsExtensionPoint.ts

```diff
@@ -83,12 +83,19 @@ interface IUserFriendlyViewDescriptor {
 
 	icon?: string;
 	contextualTitle?: string;
+	initialState?: string;
 
 	// From 'remoteViewDescriptor' type
 	group?: string;
 	remoteName?: string | string[];
 }
 
+enum InitialState {
+	Visible = 'visible',
+	Collapsed = 'collapsed',
+	Hidden = 'hidden'
+}
+
```

Comment:
Have you thought about making initialState an object type that can take `visibility` and `collapsed` states as two separate properties?

```
"hidden":  false,
"expanded": false
```

Allows to have a view that is visible and collapsed by default. This also aligns with existing default?

Reply:
Sounds good to me. I would name them `hidden` and `collapsed` or `visible` and `expanded`, so that the default would be either all `true` or all `false. Do you have a preference? (I will also take it to the API sync)

Reply:
> Allows to have a view that is visible and collapsed by default. 

The current implementation already allows this, just set the `initialState` to `collapsed`.

I thought of one concern with having `visible` and `expanded` separate: 
The NPM view is not visible by default. The user goes to the `...` menu and makes the NPM view visible. At this point, they have said that they want the view. The view should show up as visible _and_ expanded, since they have just taken explicit action that they want the view. In what scenario would we want a view to to be hidden and collapsed?

Reply:
> In what scenario would we want a view to to be hidden and collapsed?

Yeah good point. I do not think it is an useful combi to have hidden and collapsed by default. But other three combinations are useful.

| hidden | collapsed | |
| --- | --- | --- |
| true | true |  ❓ 
| true | false | ✅ 
| false | true | ✅ 
| false | false | ✅

Yeah, lets see what does API sync team suggest.

Reply:
I think `"hidden":true` and `"collapsed":false` is also not meaningful, since the view is hidden, who cased whether it is collapsed?

At the API sync, an enum was preferred. The property would be called `visibility` and could be set to `visible`(default), `hidden`, or `collapsed`. @sandy081 what do you think of those options?

Reply:
Liked it. 👍